Models/data_utils/load_data_ego_path.py

```diff
@@ -0,0 +1,208 @@
+
+        self.label_filepath = labels_filepath
+        self.image_dirpath = images_filepath
+        self.dataset_name = dataset
+
+        # ================= Preliminary checks ================= #
+
+        if not (self.dataset_name in VALID_DATASET_LIST):
+            raise ValueError("Unknown dataset! Contact our team so we can work on this.")
+        
+        # Load JSON labels, address the diffs of format across datasets
+        with open(self.label_filepath, "r") as f:
+            self.labels = json.load(f)
+        if "data" in self.labels:                   # Some has "data" parent key
+            self.labels = self.labels["data"]       # Make sure it gets the "data" part
+        # Some even stores ego path data like this:
+        # data : [
+        #   {
+        #       "00001" : ...
+        #   }
+        # ]
+        # So convert it back to dict to be compatible with the rest
+        if type(self.labels) is list:
+            self.labels = {
+                frame_code : content
+                for smaller_dict in self.labels
+                for frame_code, content in smaller_dict.items()
+            }
+
+        self.images = sorted([
+            f for f in pathlib.Path(self.image_dirpath).glob("*.png")
+        ])
+
+        self.N_labels = len(self.labels)
+        self.N_images = len(self.images)
+
+        # Sanity check func by Mr. Zain
+        checkData = CheckData(
+            self.N_images,
+            self.N_labels
+        )
+        
+        # ================= Initiate data loading ================= #
+
+        self.train_images = []
+        self.train_labels = []
+        self.val_images = []
+        self.val_labels = []
+
+        self.N_trains = 0
+        self.N_vals = 0
+
+        if (checkData.getCheck()):
+            for set_idx, frame_id in enumerate(self.labels):
+                if (set_idx % 10 == VAL_SET_FRACTION * 10):
+                    # Slap it to Val
+                    self.val_images.append(str(self.images[set_idx]))
+                    self.val_labels.append(self.labels[frame_id])
+                    self.N_vals += 1 
+                else:
+                    # Slap it to Train
+                    self.train_images.append(str(self.images[set_idx]))
+                    self.train_labels.append(self.labels[frame_id])
+                    self.N_trains += 1
+
+        print(f"Dataset {self.dataset_name} loaded with {self.N_trains} trains and {self.N_vals} vals.")
+        print(f"Val/Total = {(self.N_vals / (self.N_trains + self.N_vals))}")
+
+    # Get sizes of Train/Val sets
+    def getItemCount(self):
+        return self.N_trains, self.N_vals
+
+    # ================= Get item at index ith, returning img and EgoPath ================= #
+    
+    # For train
+    def getItemTrain(self, index):
+        img_train = Image.open(str(self.train_images[index])).convert("RGB")
+        label_train = self.train_labels[index]["drivable_path"]
+        
```

Comment:
Add a filter here to address the issue of normalized y-coordinate in CurveLanes.